packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -42,44 +45,64 @@ export interface UserPoolClientProps {
 
   /**
    * List of enabled authentication flows
-   * @default no enabled flows
+   * @default - no enabled flows
    */
   readonly enabledAuthFlows?: AuthFlow[]
 }
 
 /**
- * Define a UserPool App Client
+ * Represents a Cognito user pool client.
  */
-export class UserPoolClient extends Resource {
+export interface IUserPoolClient extends IResource {
   /**
+   * Name of the application client
    * @attribute
    */
-  public readonly userPoolClientId: string;
+  readonly userPoolClientId: string;
+}
 
+/**
+ * Define a UserPool App Client
+ */
+export class UserPoolClient extends Resource implements IUserPoolClient {
   /**
-   * @attribute
+   * Import a user pool client given its id.
    */
-  public readonly userPoolClientName: string;
+  public static fromUserPoolClientId(scope: Construct, id: string, userPoolClientId: string): IUserPoolClient {
+    class Import extends Resource implements IUserPoolClient {
+      public readonly userPoolClientId = userPoolClientId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly userPoolClientId: string;
```

Comment:
doc?

Reply:
Automatically inherited from the interface, I believe. Hence the no complaints from `awslint`.